docs/installation.rst

```diff
@@ -160,6 +160,18 @@ use ``--no-binary``.
 
     $ pip install cryptography --no-binary cryptography
 
+Cross Compiling
+~~~~~~~~~~~~~~~
+
+If you are cross compiling for a target system that has the same target triple
+as the build host you should add these variables in your environment to prevent
+cargo from linking host build-scripts against the target libraries.
+
+.. code-block:: console
+
+    __CARGO_TEST_CHANNEL_OVERRIDE_DO_NOT_USE_THIS="nightly"
```

Comment:
The `DO_NOT_USE_THIS` is kind of a red-flag that this probably isn't the right way to do this 😂

I'm not sure I understand the use case here, what does it mean to cross-compile for a target with the same triple as the host?

Reply:
> The `DO_NOT_USE_THIS` is kind of a red-flag that this probably isn't the right way to do this 😂

It's the correct way to do this on a stable rust toolchain, because the fix is behind a nightly flag and that's the only way to enable a nightly flag on the stable toolchain. By policy nightly flags are not supported/documented as available on stable toolchains but there's not technical reason that they don't won't work(they work effectively the same as they do on a nightly toolchain if you set `__CARGO_TEST_CHANNEL_OVERRIDE_DO_NOT_USE_THIS="nightly"`), this particular nightly only feature is needed to fix a critical cross compilation toolchain bug. I've [been trying to stabilize this for a while](https://github.com/rust-lang/cargo/pull/9753) but that's stalled for unclear reasons.

> I'm not sure I understand the use case here, what does it mean to cross-compile for a target with the same triple as the host?

For example if the host triple(ie `x86_64-unknown-linux-gnu`) is identical to the target triple(ie `x86_64-unknown-linux-gnu`) then you need to set this in the env when cross compiling otherwise the build will likely fail with cryptic build script linker errors due to cargo using the wrong linker for build scripts.

Reply:
Why do you want to cross compile at all if the host triple is the target triple? Wouldn't you just compile it? I'm assuming there's a nuance I'm missing here 😄

Reply:
> Why do you want to cross compile at all if the host triple is the target triple? Wouldn't you just compile it? I'm assuming there's a nuance I'm missing here 😄

For example I'm trying to cross compile a rootfs image for an embedded `x86_64` board using [buildroot](https://buildroot.org/).

Reply:
We are currently using [this fix with our infrastructure](https://github.com/buildroot/buildroot/blob/ac573c55aac3cbb4257f5388c91321c81095c654/package/pkg-cargo.mk#L26-L50) to build [python-cryptography](https://github.com/buildroot/buildroot/blob/01764876e3d4e7ab95c8365f1c83e6c12d49757c/package/python-cryptography/python-cryptography.mk#L17).

Reply:
(Apologies for not following up here more promptly.)

I guess the thing I don't understand is why this is considered cross-compilation at all? If the host and target are both "the same", what's being solved here?